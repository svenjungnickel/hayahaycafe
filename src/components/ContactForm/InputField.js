import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';

const InputField = ({ controlId, name, label, errorMessage, onChange, description, ariaLabel }) => (
    <Form.Group controlId={controlId}>
        {'' !== label && <Form.Label>{label} *</Form.Label>}
        <Form.Control name={name} type="text" required onChange={onChange} aria-label={ariaLabel} />
        {description && <Form.Text className="text-muted">{description}</Form.Text>}
        <Form.Control.Feedback type="invalid">{errorMessage}</Form.Control.Feedback>
    </Form.Group>
);

InputField.propTypes = {
    controlId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    errorMessage: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    description: PropTypes.string,
    ariaLabel: PropTypes.string,
};

export default InputField;
