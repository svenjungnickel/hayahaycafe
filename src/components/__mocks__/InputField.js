import React from 'react';

const InputField = () => <input />;

export default InputField;
