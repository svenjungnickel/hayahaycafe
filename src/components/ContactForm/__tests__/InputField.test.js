import React from 'react';
import renderer from 'react-test-renderer';
import InputField from '../InputField';

describe('InputField', () => {
    it('renders without required controlId throws prop type error', () => {
        const renderInputField = () => {
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = <InputField name={name} label={label} errorMessage={errorMessage} onChange={onChange} />;
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders without required name throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = (
                <InputField controlId={controlId} label={label} errorMessage={errorMessage} onChange={onChange} />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders without required label throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = (
                <InputField controlId={controlId} name={name} errorMessage={errorMessage} onChange={onChange} />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders without required errorMessage throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const onChange = () => {};

            const component = <InputField controlId={controlId} name={name} label={label} onChange={onChange} />;
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders without required onChange throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';

            const component = (
                <InputField controlId={controlId} name={name} label={label} errorMessage={errorMessage} />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid controlId throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 123;
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid name throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 123;
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid label throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 123;
            const errorMessage = 'errorMessage';
            const onChange = () => {};

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid errorMessage throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 123;
            const onChange = () => {};

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid onChange throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = {};

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid type throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};
            const type = 'invalid';

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                    type={type}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid description throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};
            const description = 123;

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                    description={description}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid ariaLabel throws prop type error', () => {
        const renderInputField = () => {
            const controlId = 'controlId';
            const name = 'name';
            const label = 'label';
            const errorMessage = 'errorMessage';
            const onChange = () => {};
            const ariaLabel = 123;

            const component = (
                <InputField
                    controlId={controlId}
                    name={name}
                    label={label}
                    errorMessage={errorMessage}
                    onChange={onChange}
                    ariaLabel={ariaLabel}
                />
            );
            renderer.create(component);
        };

        expect(renderInputField).toThrowError('Warning: Failed prop type');
    });

    it('renders with required parameters', () => {
        const controlId = 'controlId';
        const name = 'name';
        const label = 'label';
        const errorMessage = 'errorMessage';
        const onChange = () => {};

        const component = (
            <InputField
                controlId={controlId}
                name={name}
                label={label}
                errorMessage={errorMessage}
                onChange={onChange}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with description', () => {
        const controlId = 'controlId';
        const name = 'name';
        const label = 'label';
        const errorMessage = 'errorMessage';
        const onChange = () => {};
        const description = 'description';

        const component = (
            <InputField
                controlId={controlId}
                name={name}
                label={label}
                errorMessage={errorMessage}
                onChange={onChange}
                description={description}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders with empty label and filled ariaLabel', () => {
        const controlId = 'controlId';
        const name = 'name';
        const label = '';
        const errorMessage = 'errorMessage';
        const onChange = () => {};
        const ariaLabel = 'ariaLabel';

        const component = (
            <InputField
                controlId={controlId}
                name={name}
                label={label}
                errorMessage={errorMessage}
                onChange={onChange}
                ariaLabel={ariaLabel}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
