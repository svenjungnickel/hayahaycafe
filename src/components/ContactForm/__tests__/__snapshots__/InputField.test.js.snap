// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders with description 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="controlId"
  >
    label
     *
  </label>
  <input
    className="form-control"
    id="controlId"
    name="name"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <small
    className="text-muted form-text"
  >
    description
  </small>
  <div
    className="invalid-feedback"
  >
    errorMessage
  </div>
</div>
`;

exports[`InputField renders with empty label and filled ariaLabel 1`] = `
<div
  className="form-group"
>
  <input
    aria-label="ariaLabel"
    className="form-control"
    id="controlId"
    name="name"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <div
    className="invalid-feedback"
  >
    errorMessage
  </div>
</div>
`;

exports[`InputField renders with required parameters 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="controlId"
  >
    label
     *
  </label>
  <input
    className="form-control"
    id="controlId"
    name="name"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <div
    className="invalid-feedback"
  >
    errorMessage
  </div>
</div>
`;
